### PR TITLE
Test DNS on Windows 'nat' networks

### PR DESCRIPTION
**Pull request by robmry**
**- What I did**

Add a regression test for https://github.com/moby/moby/issues/47370

**- How I did it**

Use the Windows busybox image, ping between two containers - first on the default nat network, then on a user-defined nat network.

**- How to verify it**

New tests.

**- Description for the changelog**
```markdown changelog
Added regression test for DNS names in the default 'nat' network on Windows.
```

### CODE REVIEW

**Review by akerouanton — APPROVED** [review]
LGTM

**Review by vvoland — APPROVED** [review]
LGTM; nit can be ignored

File: integration/networking/bridge_test.go [hunk 1]
```diff
@@ -191,6 +191,66 @@ func TestBridgeICC(t *testing.T) {
...
+			defer cancel()
+			res := container.RunAttach(attachCtx, t, c,
+				container.WithName(ctr2Name),
+				container.WithCmd(pingCmd...),
+				container.WithNetworkMode(tc.netName),
+			)
+			defer c.ContainerRemove(ctx, res.ContainerID, containertypes.RemoveOptions{Force: true})
+
+			assert.Check(t, is.Equal(res.ExitCode, 0))
+			assert.Check(t, is.Equal(res.Stderr.Len(), 0))
```
[the remark this answers is not in the document]
Comment (robmry) [reply integration/networking/bridge_test.go:248]: Oh, sure - I'll do that ...

[the remark this answers is not in the document]
Comment (robmry) [reply integration/networking/bridge_test.go]: Ah, right - I'll just undo it ...

> bridge_test.go:248: assertion failed: type *bytes.Buffer does not have a length

[the remark this answers is not in the document]
Comment (vvoland) [reply integration/networking/bridge_test.go:248]: Oh, sorry!

That's weird though 🤔

[the remark this answers is not in the document]
Comment (vvoland) [reply integration/networking/bridge_test.go:248]: Hmm right, `cmp.Len` seems to use `reflect.Len` instead of just casting to `interface { Len() int}`

**Comment by thaJeztah** [comment]
Failure is unrelated (`TestDiskUsage` is a known flaky test)

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
@robmry @akerouanton I see this one was also marked for backporting

**Pull request merged**